3CodeChallenge/app/app/src/main/java/com/example/myapplication/MainActivity.java

```diff
@@ -0,0 +1,103 @@
+package com.example.myapplication;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.os.Bundle;
+import android.util.Patterns;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.CheckBox;
+import android.widget.EditText;
+import android.widget.Spinner;
+import android.widget.Toast;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MainActivity extends AppCompatActivity implements AdapterView.OnItemSelectedListener {
+
+    private EditText emailEditText, phoneEditText;
+    private CheckBox termsCheckbox;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        setContentView(R.layout.views_sample_login_view);
+        displayViewsSampleLoginViews();
+    }
+
+    public void displayViewsSampleLoginViews() {
+        emailEditText = findViewById(R.id.emailEditText);
+        phoneEditText = findViewById(R.id.phoneEditText);
+        termsCheckbox = findViewById(R.id.termsCheckbox);
+    }
+
+    public void buttonSubmit(View view) {
+        String contentFromEmailEditText = emailEditText.getText().toString();
+        String contentFromPhoneEditText = phoneEditText.getText().toString();
+
+        if (!Patterns.EMAIL_ADDRESS.matcher(contentFromEmailEditText).matches() || !Patterns.PHONE.matcher(contentFromPhoneEditText).matches() || !termsCheckbox.isChecked()) {
+            emailEditText.setError(getString(R.string.submit_error));
+        } else
+            Toast.makeText(this, "Done", Toast.LENGTH_SHORT).show();
```

Comment:
Please consider avoiding to add hardcoded strings. Define them in strings.xml. Use ALT + Enter to define them